ClientApp/src/app/services/tasks/tasks.service.ts

```diff
@@ -0,0 +1,300 @@
+import {
+    Injectable,
+    Inject
+  } from '@angular/core';
+  import { HttpClient, HttpErrorResponse } from '@angular/common/http';
+  
+  import {
+    SprintData,
+    SprintNotFoundError
+  } from 'src/types/sprint';
+  import {
+    TaskData,
+    TaskNotFoundError
+  } from 'src/types/task';
+  import {
+    ProjectData,
+    ProjectNotFoundError
+  } from 'src/types/project';
+
+  import {
+    catchError,
+    map,
+    Observable,
+    of,
+    switchMap,
+    throwError
+  } from 'rxjs';
+  
+  import { LabelData, LabelNotFoundError } from 'src/types/label';
+  import { environment } from 'src/environments/environment';
+  
+  @Injectable({providedIn: 'root'})
+  export class TaskApi {
+    baseUrl: string = environment.url;
+    http: HttpClient;
+  
+    constructor(http: HttpClient) {
+      this.http = http;
+    }
+  
+    setHttpClient(http: HttpClient) {
+      this.http = http;
+    }
+  
+    /**
+     * 
+     * @param id the id of the requested task
+     * @returns `TaskData`
+     * @throws `TaskNotFoundError` on invalid id
+     */
+    getTaskData(id: number): Observable<TaskData> {
+
+
+    return this.http.get<TaskData>(`${environment.url}Tasks/${id}`, {	
+            observe: 'body',
+            responseType: 'json'
+	});
+    }
+
+
+    //Get all tasks
+    getAllTasks(): Observable<TaskData[]> {
+        return this.http.get<TaskData[]>(`${environment.url}Tasks`, {	
```

Comment:
Is there any error checking that needs to be done on these http calls? I'm not sure what kinds of errors would turn up